### PR TITLE
Add procedure for getting a kerberos ticket with OTP enabled

### LINKED ISSUE #433: [RFE] Add warning/info text box in noggin for OTP
As found out during some tests, and discussed in #fedora-aaa, it seems that enabling OTP enforces 2FA everywhere.
So a user who has enabled it would find himself in troubles for simple `kinit`.
While we can have another card/spike for what needs to be done to have `pkinit` working externally (it doesn't actually, but it's been evaluated), this card is more to just had warning/info box in /user/<user>/settings/otp/ that would explain the `kinit` change and how to get kerberos ticket (if people want to build in koji, query fasjson, etc)

### PR DESCRIPTION
**Pull request by ryanlerch**
Resolves: fedora-infra/noggin#433

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

### CODE REVIEW

**Comment by nirik** [comment]
So, a bit of this we may be able to simplify by modifying the kerberos config we ship in fedora-packager package. 
but some questions: 
* Does this config mess up the non otp case? ie, if you set this up and have an otp enrolled and everything works, then you go and delete it and go back to just password, do you need to undo these steps, or the still work ok ?
* How does this interact with kcm? (The sssd kerberos cache store)
* How does this interact with gnome-online-accounts? 

Thanks for writing this up!

**Comment by markobrien1** [comment]
> So, a bit of this we may be able to simplify by modifying the kerberos config we ship in fedora-packager package.
> but some questions:
> 
> * Does this config mess up the non otp case? ie, if you set this up and have an otp enrolled and everything works, then you go and delete it and go back to just password, do you need to undo these steps, or the still work ok ?

I have tested this by setting my user auth in ipa to password and it works. That scenario shouldn't happen though, once you create an OTP token in noggin you can't go back to not having one. You can create a new one and delete the old but to completely remove it would require admin intervention 

> * How does this interact with kcm? (The sssd kerberos cache store)

Once you auth with this your local ticket cache *should* act the way it does at present. This is just a round about way to get the ticket

> * How does this interact with gnome-online-accounts?

We hope this works the same but we haven't tested it. What would be the best way to test this for staging?

> 
> Thanks for writing this up!

Seconded, thanks Ryan

File: modules/ROOT/pages/user.adoc [hunk 1]
```diff
@@ -45,10 +45,87 @@ scan the QR code and add it to your app.
...
+
+Install the `krb5-pkinit` package from the Fedora repositories:
+
+ sudo dnf install krb5-pkinit
+
+=== 2. Get the IPA certificate
+
+Download the certificate from Fedora Accounts
+
+ sudo curl https://admin.fedoraproject.org/accounts/ipa_ca.crt -o /etc/krb5.conf.d/fedoraproject_ipa_ca.crt
```
Comment (markobrien1) [inline modules/ROOT/pages/user.adoc]: This is the only bit holding up the PR, that url doesn't exist :) 

Also for the output location, maybe we could make this 2 steps
1. sudo mkdir /etc/pki/ipa
2. sudo curl https://admin.fedoraproject.org/accounts/ipa_ca.crt -o /etc/pki/ipa/fedoraproject_ipa_ca.crt

/etc/pki is a home for certs so make sense to add it there. 
Obviously the location would have to be updated in the following steps if you do it here :)

**Comment by ryanlerch** [comment]
> > * How does this interact with gnome-online-accounts?
> 
> We hope this works the same but we haven't tested it. What would be the best way to test this for staging?

FWIW, when i tested out this procedure, i did it on my local [tiny-stage](https://github.com/fedora-infra/tiny-stage) setup, and it all worked ok. If i get a chance, will check the gnome-online-accounts thing too.

**Comment by ryanlerch** [comment]
> So, a bit of this we may be able to simplify by modifying the kerberos config we ship in fedora-packager package.

Are we able to ship the certs in this package too? (for stage and prod), as well as changing the config files in krb5.conf.d ?
this would make this procedure a lot easier...

**Comment by nirik** [comment]
> > So, a bit of this we may be able to simplify by modifying the kerberos config we ship in fedora-packager package.
> 
> Are we able to ship the certs in this package too? (for stage and prod), as well as changing the config files in krb5.conf.d ?
> this would make this procedure a lot easier...

I think we could probibly yeah... as long as they don't change much, and I don't think they would unless we reinstalled everything or something similarly drastic.

**Comment by ryanlerch** [comment]
> I think we could probibly yeah... as long as they don't change much, and I don't think they would unless we reinstalled everything or something similarly drastic.

Okay, filed a PR to do this in fedora-packager:

https://pagure.io/fedora-packager/pull-request/166

of course the actual certs aren't in there yet, just placeholders for now do the PR can be reviewed

**Pull request merged**